webui/react/src/components/ThemeToggle.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useEffect } from 'react';
+
+import useSettings from 'hooks/useSettings';
+import useTheme from 'hooks/useTheme';
+import { DarkLight } from 'themes';
+
+import css from './ThemeToggle.module.scss';
+import { config, Settings, ThemeClass } from './ThemeToggle.settings';
+
+interface ThemeOption {
+  className: ThemeClass;
+  displayName: string;
+  next: ThemeClass
+}
+
+const ThemeOptions: {[theme: string] : ThemeOption} = {
+  [ThemeClass.LIGHT]: {
+    className: ThemeClass.LIGHT,
+    displayName: 'Light Mode',
+    next: ThemeClass.DARK,
+  },
+  [ThemeClass.DARK]: {
+    className: ThemeClass.DARK,
+    displayName: 'Dark Mode',
+    next: ThemeClass.SYSTEM,
+  },
+  [ThemeClass.SYSTEM]: {
+    className: ThemeClass.SYSTEM,
+    displayName: 'System Mode',
+    next: ThemeClass.LIGHT,
+  },
+};
+
+const ThemeToggle: React.FC = () => {
+
+  const {
+    settings,
+    updateSettings,
+  } = useSettings<Settings>(config);
```

Comment:
**suggestion:** I think it'll be better to have `useSettings` inside `useTheme`. Currently, if for whatever reason, `ThemeToggle` isn't included, the theme will not be properly pulled in from local storage (or user preferences in the near future).